### ISSUE TITLE
Control flow schematic: is it possible to keep html comments inside blocks?

### ISSUE THREAD

**Issue by mattlewis92** [angular/angular#53587]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

For example, this code:
```
<button>
  <!-- prettier-ignore -->
  <ng-container
    *ngIf="condition; else other"
    i18n="@@text"
  >
    ICU: {count, plural, =1 {item} other {items}}
  </ng-container>
  <ng-template #other>Bar</ng-template>
</button>
```

will be converted to this:
```
<button>
  <!-- prettier-ignore -->
  @if (condition) {
    <ng-container i18n="@@text">
      ICU: {count, plural, =1 {item} other {items}}
    </ng-container>
  } else {
    Bar
  }
</button>
```

However, as the comment has meaning (to ignore the next block from formatting with prettier), this now causes a change in behaviour, as the inner ng-container is no longer ignored from formatting. 

Ideally, the comment would be printed inside the condition, like this:
```
<button>
  @if (condition) {
    <!-- prettier-ignore -->
    <ng-container i18n="@@text">
      ICU: {count, plural, =1 {item} other {items}}
    </ng-container>
  } else {
    Bar
  }
</button>
```

The same problem occurs for eslint ignore comments like this `<!-- eslint-disable-next-line @angular-eslint/template/rule -->` inside templates, they are now no longer printed before the line they are meant to disable and no longer have the desired effect.

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.5
Node: 20.9.0
Package Manager: pnpm 8.12.0
OS: darwin arm64

Angular: 17.0.6+sha-1c55c49
... core

Package                             Version
-------------------------------------------------------------
@angular-devkit/architect           0.1700.6
@angular-devkit/build-angular       17.0.5
@angular-devkit/core                17.0.5
@angular-devkit/schematics          17.0.5
@angular/animations                 17.0.5
@angular/cdk                        17.0.1
@angular/cli                        17.0.5
@angular/common                     17.0.5
@angular/compiler                   17.0.5
@angular/compiler-cli               17.0.5
@angular/elements                   17.0.5
@angular/forms                      17.0.5
@angular/google-maps                15.2.9
@angular/language-service           17.0.5
@angular/localize                   17.0.5
@angular/platform-browser           17.0.5
@angular/platform-browser-dynamic   17.0.5
@angular/router                     17.0.5
@angular/service-worker             17.0.5
@schematics/angular                 17.0.5
ng-packagr                          17.0.2
rxjs                                7.8.0
typescript                          5.2.2
webpack                             5.88.2
zone.js                             0.14.2
```

**Comment by jessicajaniuk** [comment]
TL;DR, This is probably not possible as we have no idea what those comments do, whether they're load bearing or not, and whether they should be inside or outside of a block. It's unsafe for us to touch them. Comments in this case could be anything from something like prettier notes to some sort of todo. There are far too many possible shapes that the template code could be doing for us to know where that comment should end up.

**Comment by mattlewis92** [comment]
Makes perfect sense, thank you for the detailed explanation!
